pkg/splunk/util/secrets.go

```diff
@@ -495,14 +495,13 @@ func ApplyNamespaceScopedSecretObject(ctx context.Context, client splcommon.Cont
 }
 
 // GetSecretByName retrieves namespace scoped secret object for a given name
-func GetSecretByName(ctx context.Context, c splcommon.ControllerClient, cr splcommon.MetaObject, name string) (*corev1.Secret, error) {
+func GetSecretByName(ctx context.Context, c splcommon.ControllerClient, namespace string, crName string, name string) (*corev1.Secret, error) {
```

Comment:
`crName` is just used for logging?

Reply:
Yes, initially the API `GetSecretByName` had a parameter `cr splcommon.MetaObject` passed in only for logging. I have introduced the parameter `crName` to retain the logging functionality(perhaps I can change the name of the parameter to loggingName to make it more clear).

I have changed the implementation of the API to not use `splcommon.MetaObject` as the purpose of the API is to retrieve a secret by its `name`, `namespace`(has nothing to do with `splcommon.MetaObject`. This helps in making the API being accessible to areas of code where there are no objects of type `splcommon.MetaObject` involved.

Reply:
Sounds good to me.